### PR TITLE
System modes update

### PR DESCRIPTION
**Pull request by norro**
Some minor updates in the description (and image) of the system modes concept.

### CODE REVIEW

**Comment by norro** [comment]
@ralph-lange I went for our recent and more concrete architecture image instead of the previously quite abstract architecture image. What do you think?

File: _docs/concepts/client_library/system_modes/index.md [hunk 1]
```diff
@@ -37,16 +38,17 @@ Our observation is that interweaving of task handling, contingency handling, and
...
 
-The envisioned key elements to achieve this goal are:
+The key elements of the approach are:
 
-1.  Extensible concept to specify the runtime states of components, i.e ROS 2 nodes.
-2.  Modeling approach for specifying system modes based on these component states.
-3.  Diagnosis module for deriving relevant information from the operating systems, the hardware and the functional components.
-4.  Mode manager module for system runtime configuration.
+1.  _Extended Lifecycle_: Extensible concept to specify the runtime states of components, i.e ROS 2 lifecycle nodes.
+2.  _System Hierarchy and Modes_: Modeling approach for specifying system modes based on these component states.
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md]: Explain the hierarchy model in some words, not only modes.

File: _docs/concepts/client_library/system_modes/index.md [hunk 2]
```diff
@@ -81,7 +83,17 @@ A simple example is provided at:
 [github.com:system_modes_examples/README.md#example-mode-file](https://github.com/micro-ROS/system_modes/blob/master/system_modes_examples/README.md#example-mode-file)
 
 
-### Mode manager
+### Mode Inference
+
+The mode inference infers the entire system states (and modes) based on the lifecycle states, modes, and parameter configuration of its components, i.e. the ROS 2 lifecyle nodes. It parses the model and subscribes to lifecycle/mode change requests, lifecycle/mode changes, and parameter events.
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md]: Give some intuition for the inference mechanisms. Most important the differentiation between actual states and parameters of the nodes, inferred modes of the (sub)systems and desired states/modes?

File: _docs/concepts/client_library/system_modes/index.md [hunk 3]
```diff
@@ -81,7 +83,17 @@ A simple example is provided at:
...
+
+The documentation and code can be found at:  
+[github.com:system_modes/README.md#mode-inference](https://github.com/micro-ROS/system_modes/blob/master/system_modes/README.md#mode-inference)  
+The mode inference can be best observed in the mode monitor, a console-based debugging tool, see:  
+[github.com:system_modes/README.md#mode-monitor](https://github.com/micro-ROS/system_modes/blob/master/system_modes/README.md#mode-monitor)
+
+
+### Mode Manager
 
 The mode manager allows for runtime system adaptation based on such a system hierarchy and modes model. It parses the model and provides all services and topics to request state and mode changes and to monitor these changes.
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md]: I propose to add a sentence that the mode manager includes the mode inference mechanisms. Then you may add in the last sentences of this paragraph something like: "in addition to the topics and events monitored by the inference mechanisms, the mode manager provides all services ..."

File: _docs/concepts/client_library/system_modes/index.md (hunk 1 above)
Comment (ralph-lange) [reply _docs/concepts/client_library/system_modes/index.md]: In the whole text, I please differentiate between the system hierarchy model and the modes more explicitly.

**Pull request merged**